flight/Modules/Actuator/actuator.c

```diff
@@ -1217,12 +1217,11 @@ static void smithp_compensate(struct smith_predictor *m, float *desired_vect)
 			region, that makes throttle eventually oscillate around the zero point under certain conditions.
 			This leads to funny business with the motors, e.g. grinding. */
 			float t = desired_vect[i];
-			if ((t >= THROTTLE_EPSILON && (t+v) < THROTTLE_EPSILON) ||
-			    (t <= -THROTTLE_EPSILON && (t+v) > -THROTTLE_EPSILON)) {
-					v = 0;
-				}
-			/* Also bound throttle. */
-			desired_vect[i] = bound_sym(desired_vect[i] + v, 1.0f);
+			float r = t+v;
+
+			/* Don't cross zero via prediction. Also bound throttle. */
+			if (sign(t) == sign(r))
```

Comment:
THis isn't quite what we want, because if you overperform significantly and throttle is 10%, and it wants to lower it to -2%... we should still lower it to throttle epsilon.